Detectors/TPC/reconstruction/src/TPCCATracking.cxx

```diff
@@ -0,0 +1,156 @@
+    const int rowInSector = cluster.getRow() + region.getGlobalRowOffset();
+    const float padY = cluster.getPadMean();
+    const int padNumber = int(padY);
+    const GlobalPadNumber pad = mapper.globalPadNumber(PadPos(rowInSector, padNumber));
+    const PadCentre& padCentre = mapper.padCentre(pad);
+    const float localY = padCentre.Y() - (padY - padNumber - 0.5) * region.getPadWidth();
+    const float localYfactor = (cru.side() == Side::A) ? -1.f : 1.f;
+    float zPositionAbs = cluster.getTimeMean()*elParam.getZBinWidth()*gasParam.getVdrift();
+    if (!mTrackingCAO2Interface->GetParamContinuous()) zPositionAbs = detParam.getTPClength() - zPositionAbs;
+
+    Point2D<float> clusterPos(padCentre.X(), localY);
+
+    // sanity checks
+    if (zPositionAbs < 0 || (!mTrackingCAO2Interface->GetParamContinuous() && zPositionAbs > detParam.getTPClength())) {
+      LOG(INFO) << "Removing cluster " << icluster << "/" << inputClusters->GetEntries() << " time: " << cluster.getTimeMean() << ", abs. z: " << zPositionAbs << "\n";
+      continue;
+    }
+
+    hltCluster.fId = icluster;
+    hltCluster.fX = clusterPos.X();
+    hltCluster.fY = clusterPos.Y() * (localYfactor);
+    hltCluster.fZ = zPositionAbs * (-localYfactor);
+    hltCluster.fRow = rowInSector;
+    hltCluster.fAmp = cluster.getQmax();
+
+    cd.SetNumberOfClusters(cd.NumberOfClusters() + 1);
+    nClustersConverted++;
+  }
+  if (inputClusters->GetEntries() != nClustersConverted) {
+    LOG(INFO) << "Passed " << nClustersConverted << " (out of " << inputClusters->GetEntries() << ") clusters to CA tracker\n";
+  }
+
+  retVal = mTrackingCAO2Interface->RunTracking(mClusterData, tracks, nTracks, trackClusterIDs);
+  if (retVal == 0)
+  {
+    for (int i = 0; i < nTracks; i++) {
+      if (!tracks[i].OK()) continue;
+      TrackTPC trackTPC(
+        tracks[i].GetParam().GetX(), tracks[i].GetAlpha(),
+        { tracks[i].GetParam().GetY(), tracks[i].GetParam().GetZ(), tracks[i].GetParam().GetSinPhi(),
+          tracks[i].GetParam().GetDzDs(), tracks[i].GetParam().GetQPt() },
+        { tracks[i].GetParam().GetCov(0), tracks[i].GetParam().GetCov(1), tracks[i].GetParam().GetCov(2),
+          tracks[i].GetParam().GetCov(3), tracks[i].GetParam().GetCov(4), tracks[i].GetParam().GetCov(5),
+          tracks[i].GetParam().GetCov(6), tracks[i].GetParam().GetCov(7), tracks[i].GetParam().GetCov(8),
+          tracks[i].GetParam().GetCov(9), tracks[i].GetParam().GetCov(10), tracks[i].GetParam().GetCov(11),
+          tracks[i].GetParam().GetCov(12), tracks[i].GetParam().GetCov(13), tracks[i].GetParam().GetCov(14) });
+      for (int j = 0; j < tracks[i].NClusters(); j++) {
+        trackTPC.addCluster(*(static_cast<Cluster*>(inputClusters->At(trackClusterIDs[tracks[i].FirstClusterRef() + j]))));
+      }
+      outputTracks->push_back(trackTPC);
+    }
+  }
+  mTrackingCAO2Interface->Cleanup();
+#endif
+
+  return (retVal);
```

Comment:
Hi,

that is basically not necessary, since the conditional compilation disables the initialization, returning a failure code. Therefore, it will not run the tracking, returned an uninitialized failure.

Still, in order to be more explicit, I have changed this, such that the failure code is immediately visible in the code.